uaclient/contract.py

```diff
@@ -504,11 +517,69 @@ def process_entitlements_delta(
                     "Unexpected error processing contract delta for {name}:"
                     " {delta}".format(name=name, delta=new_entitlement)
                 )
+        else:
+            if service_enabled:
```

Comment:
This logic can be problematic if the service transitions to unentitled as can be seen [here](https://github.com/canonical/ubuntu-pro-client/blob/main/uaclient/entitlements/base.py#L1054)